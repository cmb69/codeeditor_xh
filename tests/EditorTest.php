<?php

namespace Codeeditor;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\View;

class EditorTest extends TestCase
{
    public function testIncludesEditor(): void
    {
        global $hjs;

        $hjs = "";
        $sut = new Editor("../", "night", "", $this->view());
        $sut->doInclude();
        Approvals::verifyHtml($hjs);
    }

    public function testInitializesEditor(): void
    {
        global $hjs, $bjs;

        $hjs = $bjs = "";
        $sut = new Editor("../", "default", "", $this->view());
        $sut->init();
        Approvals::verifyHtml($bjs);
    }

    public function testReplacesTextarea(): void
    {
        $sut = new Editor("../", "default", "", $this->view());
        Approvals::verifyHtml($sut->replace("my_textarea", ""));
    }

    public function testFindsCodeMirrorThemes(): void
    {
        $sut = new Editor("../", "default", "", $this->view());
        $this->assertContains("night", $sut->getThemes());
    }

    private function view(): View
    {
        return new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["codeeditor"]);
    }
}
