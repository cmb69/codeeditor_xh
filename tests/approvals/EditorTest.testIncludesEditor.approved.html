
<link rel="stylesheet" type="text/css" href="../codeeditor/codemirror/codemirror-combined.css">
<link rel="stylesheet" type="text/css" href="../codeeditor/codemirror/theme/night.css">
<script src="../codeeditor/codemirror/codemirror-compressed.js"></script>
<script src="../codeeditor/codeeditor.min.js"></script>
<script>
codeeditor.text = {&quot;confirmLeave&quot;:&quot;Your modifications will be lost!&quot;};
</script>
